### ISSUE TITLE
Type mismatch in FixProcessor._addFix_undefinedMethod_create

### ISSUE THREAD

**Issue by CaiJingLong** [dart-lang/sdk#35491]
Analyzer Feedback from IntelliJ

# Version information

- `IDEA AI-181.5540.7.32.5056338`
- `2.1.0-dev.9.4`
- `AI-181.5540.7.32.5056338, JRE 1.8.0_152-release-1136-b06x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.13.6 unknown, screens 2048x1152  Retina`

# Exception

Please append the contents of:
file:///private/var/folders/_w/_rx69q2x10z2_4zvl0vfsmcm0000gn/T/report.txt
[report.txt](https://github.com/dart-lang/sdk/files/2706750/report.txt)

```
Dart analysis server, SDK version 2.1.0-dev.9.4, server version 1.21.1, FATAL error: Failed to handle request: {id: 1094, method: edit.getFixes, params: {file: /private/tmp/key_example/lib/error2/mixin2.dart, offset: 41}, clientRequestTime: 1545634430313}
type 'MixinDeclarationImpl' is not a subtype of type 'ClassDeclaration'
#0      FixProcessor._addFix_undefinedMethod_create (package:analysis_server/src/services/correction/fix_internal.dart:3315:9)
<asynchronous suspension>
#1      FixProcessor.compute (package:analysis_server/src/services/correction/fix_internal.dart:535:13)
<asynchronous suspension>
#2      DefaultFixContributor.internalComputeFixes (package:analysis_server/src/services/correction/fix_internal.dart:89:41)
<asynchronous suspension>
#3      EditDomainHandler._computeServerErrorFixes (package:analysis_server/src/edit/edit_domain.dart:580:49)
<asynchronous suspension>
#4      EditDomainHandler.getFixes (package:analysis_server/src/edit/edit_domain.dart:252:32)
<asynchronous suspension>
#5      EditDomainHandler.handleRequest (package:analysis_server/src/edit/edit_domain.dart:360:9)
#6      AnalysisServer.handleRequest.<anonymous closure>.<anonymous closure> (package:analysis_server/src/analysis_server.dart:645:45)
#7      _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:258:15)

**Comment by DanTup** [comment]
This looks similar to #34809 (albeit in a different place), so may be a similar fix.

It was reported by a Dart Code user here too:

https://github.com/Dart-Code/Dart-Code/issues/1367

**Comment by CaiJingLong** [comment]
I just found this problem in the process of using it. I can't remember exactly which step appeared. After Bao made a mistake

**Comment by scheglov** [comment]
I think you hit the problem that was fixed in 2784473939b664efa46863ad85147745447ba796.
